### ISSUE TITLE
Adding an existing script to a node which has the same case-insensitive name warns for case mismatch

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#31995]
Godot 3.2 master ccf294b92f02af6e60206e220dcf4a8474f73f0a

I have an existing script named `interaction.gd` and a node called `Interaction`. I used the right-click menu to add that script. The dialog popped up proposing the name `Interaction.gd`. I clicked on the folder icon to pick `interaction.gd` from the file system, then clicked `OK`.

Later I noticed the following warnings in the console:
```
WARNING: FileAccessWindows::_open: Case mismatch opening requested file 'Interaction.gd', stored as 'interaction.gd' in the filesystem. This file will not open when exported to other case-sensitive platforms.
     At: drivers\windows\file_access_windows.cpp:105
WARNING: FileAccessWindows::_open: Case mismatch opening requested file 'Interaction.gd', stored as 'interaction.gd' in the filesystem. This file will not open when exported to other case-sensitive platforms.
     At: drivers\windows\file_access_windows.cpp:105
Loading resource: res://dmc_terrain/Interaction.gd
WARNING: FileAccessWindows::_open: Case mismatch opening requested file 'Interaction.gd', stored as 'interaction.gd' in the filesystem. This file will not open when exported to other case-sensitive platforms.
     At: drivers\windows\file_access_windows.cpp:105
WARNING: FileAccessWindows::_open: Case mismatch opening requested file 'Interaction.gd', stored as 'interaction.gd' in the filesystem. This file will not open when exported to other case-sensitive platforms.
     At: drivers\windows\file_access_windows.cpp:105
```

Although the script loaded correctly, these warnings seem to hint a problem with this sequence of actions.

**Comment by sparkart** [comment]
I don't think that it's a bug, considering it's a warning and not an error. Linux's filesystem has case sensitivity. Maybe ignoring case mismatch on Windows would make sense, since it doesn't have case sensitivity, but I feel like that wouldn't be a great idea especially if you're working in a team where other people use Linux.

**Comment by Zylann** [comment]
Read my issue again:

> he script already existed and I didn't rename it.

> It also happened during file opening, so somehow Godot tried to open the script using the wrong name.

Something definitely went wrong here :p

**Comment by sparkart** [comment]
If you told it to load interaction.gd and not Interaction.gd... I would say it's a bug.

When you clicked on the folder icon to load the script... did you actually click on "interaction.gd" to change the "file value". If you click on it.. it should change it to lowercase, if not, it'll leave the field to uppercase.

**Comment by Zylann** [comment]
After testing again, it does replace the line edit with the lowercase version `interaction.gd`.
However I noticed that it tried to load `Interaction.gd` when the file dialog opens, before I actually select the file...

**Comment by codecustard** [comment]
The error is triggered by the following:
https://github.com/codecustard/godot/blob/24e1039eb6fe32115e8d1a62a84965e9be19a2ed/editor/editor_file_dialog.cpp#L255-L256

`_request_single_thumbnail(get_current_dir().plus_file(get_current_file()));` eventually checks if file exists, which does the test by opening the file.

It looks like the line can be removed because it doesn't seem to contribute to anything.

**Comment by KoBeWi** [comment]
Still valid in 3.2.3 rc4
On master when creating script it doesn't have name for some reason, so it's impossible to test.

**Comment by ODtian** [comment]
Still valid in 4.0.2 release. 
It seems generates a upper-case naming file and causing `Class "Xxx" hides a global script class.` because there are actually 2 files loaded but one is invisible.

**Comment by CsloudX** [comment]
![image](https://github.com/user-attachments/assets/13622991-1bc9-4f9a-8387-ee22e8ce4bd5)
Everytime it report case mismatch when I start my project. and I don't know how to fix this.
